src/interceptors/ClientRequest/NodeClientRequest.test.ts

```diff
@@ -138,7 +138,7 @@ test('performs the request as-is given resolver returned no mocked response', (d
 test('emits the ENOTFOUND error connecting to a non-existing hostname given no mocked response', (done) => {
   const emitter = new AsyncEventEmitter<HttpRequestEventMap>()
   const request = new NodeClientRequest(
-    normalizeClientRequestArgs('http:', 'http://non-existing-url.com'),
+    normalizeClientRequestArgs('http:', 'http://dummy'),
```

Comment:
The actual issue here is that we are making a side-effect that is not managed by this test. I've tried migrating most of such things to use a locally spawned server but, apparently, I've missed a few tests. 

Let's spawn a server and route the requests there:

https://github.com/mswjs/interceptors/blob/98e64aa75c877d57da28a79d5939d28b5fbc0a66/test/modules/http/intercept/http.get.test.ts#L11-L15

For the test cases when we don't want the test to hit the endpoint, I'm usually returning `res.status(500)` from the actual server route so that even if the intention breaks, the response is not what the test expects.

Reply:
We want `ENOTFOUND` error in that test case. Will calling `localhost:xxx` give that error?

Reply:
Did you discover that `https://non-existing-url.com/` exist? I don't think it does, at least when I try to access it. Could it been some hiccup during local testing? I'd like to learn more, maybe we don't need this change.

Reply:
We are getting `EAI_AGAIN` on CI right now because `http://dummy` is not a valid DNS record. I hope if we revert it back to the previous URL, it will produce connection error, and the test will pass. Maybe something else interfered while you were testing.

Reply:
Well thats flakey

Reply:
How do you think we could improve this? We need to connect to a non-existing address to assert how NodeClientRequest behaves in this scenario.

Reply:
I think there is no way around it.